### PR TITLE
ci: Update the release process for v1 release

### PR DESCRIPTION
**Pull request by netroy**
1. release nightly `1.0.0-rc` from `release-v1` branch.
2. Allow creating `major` release PRs.
3. Remove the `merged back into master` step. it doesn't work.

### CODE REVIEW

File: .github/workflows/release-publish.yml [hunk 1]
```diff
@@ -55,12 +55,12 @@ jobs:
...
-      - name: Merge Release into 'master'
-        run: |
-          git fetch origin
-          git checkout --track origin/master
-          git config user.name "Jan Oberhauser"
-          git config user.email jan.oberhauser@gmail.com
-          git merge --ff n8n@${{env.RELEASE}}
-          git push origin master
-          git push origin :${{github.event.pull_request.base.ref}}
+      # - name: Merge Release into 'master'
```
Comment (krynble) [inline .github/workflows/release-publish.yml:58]: Why is it necessary to remove this part of the code?

Comment (netroy) [reply .github/workflows/release-publish.yml:58]: this has never worked. commenting this out until I can figure out how o make this work. 
I could instead delete this, and try to do this step in a n8n workflow instead.

**Review by krynble — APPROVED** [review]
LGTM

**Comment by janober** [comment]
Got released with [`n8n@0.232.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.232.0)

**Pull request merged**